integration/pkg/contracttransmitter/evm_contract_transmitter.go

```diff
@@ -55,12 +53,11 @@ func NewEVMContractTransmitterFromRPC(ctx context.Context, lggr logger.Logger, c
 		return nil, err
 	}
 
-	id, err := chainselectors.GetChainIDFromSelector(chainSelector)
+	// Get chain ID from the RPC itself as chainselector can be virtual chain
+	chainIDInt, err := client.ChainID(ctx)
```

Comment:
As discussed with @0xAustinWang we might have nonce conflict here if we are not using different private keys per selector here